nucypher/crypto/kits.py

```diff
@@ -55,11 +54,55 @@ def __init__(self,
                  ciphertext=None,
                  signature=NOT_SIGNED) -> None:
 
+        if type(capsule) == bytes:
+            raise Exception
+
         self.ciphertext = ciphertext
         self.capsule = capsule
         self.sender_verifying_key = sender_verifying_key
         self._signature = signature
 
+        self._cfrags = set()
+        self._delegating_key = None
+        self._receiving_key = None
+        self._verifying_key = None
+
+    def attach_cfrag(self, cfrag: VerifiedCapsuleFrag):
```

Comment:
Type annotations for inputs and outputs will look nice here 😄

Reply:
Fixed